internal/cri/server/restart.go

```diff
@@ -106,6 +107,14 @@ func (c *criService) recover(ctx context.Context) error {
 		metadata := sandboxstore.Metadata{}
 		err := sbx.GetExtension(podsandbox.MetadataKey, &metadata)
 		if err != nil {
+			if errors.Is(err, errdefs.ErrNotFound) {
+				err = c.client.SandboxStore().Delete(ctx, sbx.ID)
```

Comment:
Can we add error log here? Something like "could not recover sandbox X: missing metadata key"

Reply:
nod, need to log the error message that was being returned below "failed to get metadata for stored sandbox %q: %w", sbx.ID, err  to this err not found case so we always get that as output when we run into it..